### ISSUE TITLE
network::bluecoat::snmp::plugin --mode cpu

### ISSUE THREAD

**Issue by olafmarcos** [centreon/centreon-plugins#1225]
I can get correct values from CLI, but from Centreon always getting "OK: Buffer creation...."

centreon_plugins.pl --plugin=network::bluecoat::snmp::plugin --version
Plugin Version: 1.0

centreon_plugins.pl --plugin=network::bluecoat::snmp::plugin --mode=cpu --version
Mode Version: 1.0

Thanks in advance,

**Comment by garnier-quentin** [comment]
Could you connect on "centreon_storage" database and provide us the result of following SQL request:
```
select command_line from services where command_line like '%bluecoat%cpu%';
```

**Comment by olafmarcos** [comment]
No result.


My command definition (working with all other plugins and with other modes of bluecoat plugin):

"$USER1$/centreon-plugins/centreon_plugins.pl  --hostname=$HOSTADDRESS$ --snmp-version=$_HOSTSNMPVERSION$ --snmp-community=$_HOSTSNMPCOMMUNITY$ --plugin=$ARG1$ --mode=$ARG2$ $ARG3$"

The arguments:
ARG1 = network::bluecoat::snmp::plugin
ARG2 = cpu
ARG3 = --warning 80 --critical 90

My databases;
mysql> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| centreon2          |
| centreon2_storage  |
| mysql              |
| nagios             |

On "centreon2_storage":
mysql> select command_line from services where command_line is not null limit 10;
Empty set (0.00 sec)

mysql> select check_command from services where check_command is not null limit 10;
Empty set (0.00 sec)

From "centreon" database;

+------------+--------------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+--------------+--------------+----------+------------+-----------------+
| command_id | connector_id | command_name           | command_line                                                                                                                                                                        | command_example                                                     | command_type | enable_shell | graph_id | cmd_cat_id | command_comment |
+------------+--------------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+--------------+--------------+----------+------------+-----------------+
|        671 |         NULL | check_centreon_plugins | $USER1$/centreon-plugins/centreon_plugins.pl  --hostname=$HOSTADDRESS$ --snmp-version=$_HOSTSNMPVERSION$ --snmp-community=$_HOSTSNMPCOMMUNITY$ --plugin=$ARG1$ --mode=$ARG2$ $ARG3$ | !network::f5::bigip::plugin!virtualserver-status!--filter-name=VIP- |            2 |            0 |        7 |       NULL |                 |
+------------+--------------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+--------------+--------------+----------+------------+-----------------+

mysql> select * from service where command_command_id=671;
+------------+-------------------------------+--------------------+------------------+---------------------+-------------------+---------------------------------+---------------+--------------+---------------------+----------------------------+-------------------------------+------------------------------+-------------------------------+--------------------------------+---------------+---------------------------+-----------------------------+-------------------------+-----------------------------+-------------------------------+----------------------------+-----------------------------+--------------------------------+---------------------------+-----------------------------------+--------------------------------------+-------------------------------+------------------------------+-------------------------------+----------------------------------+--------------------------+-----------------+-------------------------------------------------------------------------------------------------------------------+-------------------------+----------------+------------------+------------------+
| service_id | service_template_model_stm_id | command_command_id | timeperiod_tp_id | command_command_id2 | timeperiod_tp_id2 | service_description             | service_alias | display_name | service_is_volatile | service_max_check_attempts | service_normal_check_interval | service_retry_check_interval | service_active_checks_enabled | service_passive_checks_enabled | initial_state | service_parallelize_check | service_obsess_over_service | service_check_freshness | service_freshness_threshold | service_event_handler_enabled | service_low_flap_threshold | service_high_flap_threshold | service_flap_detection_enabled | service_process_perf_data | service_retain_status_information | service_retain_nonstatus_information | service_notification_interval | service_notification_options | service_notifications_enabled | service_first_notification_delay | service_stalking_options | service_comment | command_command_id_arg                                                                                            | command_command_id_arg2 | service_locked | service_register | service_activate |
+------------+-------------------------------+--------------------+------------------+---------------------+-------------------+---------------------------------+---------------+--------------+---------------------+----------------------------+-------------------------------+------------------------------+-------------------------------+--------------------------------+---------------+---------------------------+-----------------------------+-------------------------+-----------------------------+-------------------------------+----------------------------+-----------------------------+--------------------------------+---------------------------+-----------------------------------+--------------------------------------+-------------------------------+------------------------------+-------------------------------+----------------------------------+--------------------------+-----------------+-------------------------------------------------------------------------------------------------------------------+-------------------------+----------------+------------------+------------------+
|       1843 |                           805 |                671 |             NULL |                NULL |              NULL | F5-BIGIP_Virtual-Servers_Status | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::f5::bigip::snmp::plugin!virtualserver-status!--filter-name=VIP- --threshold-overload='vs,OK,^((blue)$)' | NULL                    |              0 | 1                | 1                |
|       1857 |                           805 |                671 |             NULL |                NULL |              NULL | F5-BIGIP_Node_Status            | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::f5::bigip::snmp::plugin!node-status!--filter-name=                                                      | NULL                    |              0 | 1                | 1                |
|       1861 |                           805 |                671 |             NULL |                NULL |              NULL | F5-BIGIP_Pool_Status            | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::f5::bigip::snmp::plugin!pool-status!--filter-name=                                                      | NULL                    |              0 | 1                | 1                |
|       1862 |                           805 |                671 |             NULL |                NULL |              NULL | F5-BIGIP_Failover               | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::f5::bigip::snmp::plugin!failover!--warning-sync-status='%{syncstatus} =~ /needManualSync/'              | NULL                    |              0 | 1                | 1                |
|       1863 |                           805 |                671 |             NULL |                NULL |              NULL | F5-BIGIP_TMM-Usage              | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::f5::bigip::snmp::plugin!tmm-usage!--filter-name                                                         | NULL                    |              0 | 1                | 1                |
|       2443 |                           805 |                671 |             NULL |                NULL |              NULL | F5-BIGIP_Virtual-Servers_Status | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::f5::bigip::snmp::plugin!virtualserver-status!--threshold-overload='vs,OK,^((blue)$)'                    | NULL                    |              0 | 1                | 1                |
|       2444 |                           805 |                671 |             NULL |                NULL |              NULL | F5-BIGIP_Node_Status            | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::f5::bigip::snmp::plugin!node-status!--threshold-overload='node,OK,^((blue)$)'                           | NULL                    |              0 | 1                | 1                |
|       2455 |                            28 |                671 |             NULL |                NULL |              NULL | CPU_BlueCoat                    | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::bluecoat::snmp::plugin!cpu!--warning 80 --critical 90                                                   | NULL                    |              0 | 1                | 1                |
|       2456 |                            30 |                671 |             NULL |                NULL |              NULL | MEMORIA_BlueCoat                | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::bluecoat::snmp::plugin!memory!--warning 80 --critical 90                                                | NULL                    |              0 | 1                | 1                |
|       2457 |                            33 |                671 |             NULL |                NULL |              NULL | DISCO_BlueCoat                  | NULL          | NULL         | 2                   |                       NULL |                          NULL |                         NULL | 2                             | 2                              | NULL          | 2                         | 2                           | 2                       |                        NULL | 2                             |                       NULL |                        NULL | 2                              | 2                         | 2                                 | 2                                    |                          NULL | NULL                         | 2                             |                             NULL | NULL                     | NULL            | !network::bluecoat::snmp::plugin!disk!--warning 80 --critical 90                                                  | NULL                    |              0 | 1                | 1                |
+------------+-------------------------------+--------------------+------------------+---------------------+-------------------+---------------------------------+---------------+--------------+---------------------+----------------------------+-------------------------------+------------------------------+-------------------------------+--------------------------------+---------------+---------------------------+-----------------------------+-------------------------+-----------------------------+-------------------------------+----------------------------+-----------------------------+--------------------------------+---------------------------+-----------------------------------+--------------------------------------+-------------------------------+------------------------------+-------------------------------+----------------------------------+--------------------------+-----------------+-------------------------------------------------------------------------------------------------------------------+-------------------------+----------------+------------------+------------------+

**Comment by garnier-quentin** [comment]
Could you send me an email on qgarnier@centreon.com ? To work directly with you if you want.

**Comment by garnier-quentin** [comment]
Could you test it with the last git version ?

**Comment by garnier-quentin** [comment]
Fix b0c6a93d2c3f775cb21b51163dddbe18def7792e
